datalad/distributed/ora_remote.py

```diff
@@ -846,23 +847,27 @@ def _local_io(self):
         #return self.store_base_path.is_dir()
         return not self.storage_host
 
-    def _info(self, msg):
+    def debug(self, msg):
+        # Annex prints just the message, so prepend with
+        # a "DEBUG" on our own.
+        self.annex.debug("ORA-DEBUG: " + msg)
 
-        if self.can_notify:
+    def message(self, msg):
+        try:
             self.annex.info(msg)
-        # TODO: else: if we can't have an actual info message, at least have a
-        #       debug message.
-        #       This probably requires further refurbishment of datalad's
-        #       capability to deal with such aspects of the
-        #       special remote protocol when parsing annex' output.
+        except ProtocolError:
+            # INFO not supported by annex version.
```

Comment:
IMHO it is better to just compare  current git annex version against known once and behave accordingly. That will later allow to prune such conditional  code when our minimal supported annex version goes ahead like did recently in one of prs.

Reply:
I don't agree. It's a matter of protocol extensions which git-annex is build with. And `annexremote` asks annex about supported extensions during PREPARE. So it's not exactly a version number this is about.

Reply:
It's explicitly part of the protocol to figure supported features rather than relying on some reported version number. Seems the cleanest way to me, to make use of that. We have that specific information available in `super` anyway.

Reply:
oh, my bad.  I thought that `INFO` is like `DEBUG` just was introduced as a part of the standard protocol, and not as an EXTENSION.  I do not think it would be really "annex build dependent" but indeed better to not just rely on the version.  I guess this should be ok in the short run.
In the long run -- I feel that AnnexRemote should provide public API to query for extensions. Could you please raise an issue on https://github.com/Lykos153/AnnexRemote/issues ?  It would also be useful to alert author that API is used by 3rd party (us) and ideally should not be too fluid ;)

Reply:
I can do that, but I don't really see the benefit. What's wrong with catching that exception instead of duplicating what `annexremote.Master.info` does? (https://github.com/Lykos153/AnnexRemote/blob/master/annexremote/annexremote.py#L1256)

Reply:
re benefit: just catching a generic ProtocolError is a guess work.
API breakage on AnnexRemote side is that -- API breakage.  See above why letting developer know about it could be useful.

Reply:
>  It's just that I adapted our code to meet the paradigm annexremote is suggesting (use those methods and catch a ProtocolError)

oh, could you please point to such suggestion?

Reply:
1. Misunderstanding, I think. Not opposing letting `annexremote`'s author know, that API breakage has consequences. ;)
My point is about making extensions public.

2. 
> just catching a generic ProtocolError is a guess work.
```
        Raises
        ----------
        ProtocolError
            If INFO is not available in this version of git-annex.
```
I don't know what's unspecific about that. There's nothing else that could raise a `ProtocolError` when we call this method.

The alternative you're proposing by requesting a public extensions query is to simply reimplement `annexremote.Master.info` ourselves (see link above). Why would we rather rely on the stability of that query than the stability of `info` raising that Error?
Plus: As mentioned in original post and commit message with that approach we would then also need to either raise dependency version for `annexremote` or check its version in addition. Whereas with the exception it just works with both versions and leaves to `annexremote` what it already implemented.

> oh, could you please point to such suggestion?

Well, that's just the way it's implemented. I take code as a suggestion ;-)

Reply:
(my two cents) Catching a `ProtocolError` here looks like a good approach.

Reply:
ok, thank you @bpoldrack for the details and bearing with me.  Let's hope that in the future this generic (and base of `UnsupportedRequest` which IMHO would have been a better choice) ProtocolError will also not be spitted out for anything else while issuing an info message.  ATM it is unlikely indeed we would hit any other functionality while issuing an INFO msg.

Re API breakage "ping" -- please make friends or I will! ;)

Reply:
>  Re API breakage "ping" -- please make friends or I will! ;)

Nooo - what a mean threat! :-D

Done.